### ISSUE TITLE
TokenCacheExtensions not available in UWP

### ISSUE THREAD

**Issue by Dark-Bond** [AzureAD/microsoft-authentication-library-for-dotnet#569]
Using Assembly Microsoft.Identity.Client, Version=1.1.2.0, the class TokenCacheExtensions is available to my WPF (.NET Framework 6.1) application.  The same thing can be found in the sample application at https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2.git.  However, when I load the same version of the library into my UWP application, this class can't be found.  I have no way to persist my credentials in UWP because the wrapped class TokenCache is sealed (for some reason) and these extension methods are unavailable.

**Comment by SomkaPe** [comment]
TokenCacheExtensions  is meant to be used for cache persistence on platforms were we do not provide default persistence.
On UWP cache is stored as application data, so there is no need to implement persistence.

**Comment by Dark-Bond** [comment]
Then tell me how I clear the persistence when I want a new user to log in, please?

**Comment by SomkaPe** [comment]
Just use interactive acquire token api. When you call silent acquire token api you have to provide User, which is already logged in (by calling getUsers api you can get list of currently logged in users).

**Comment by jmprieur** [comment]
@Dark-Bond 
- there are several ways. If this is for testing only, you can un-install the UWP app, which will remove all associated data including the cache. I mention that because if you are trying to loggin a user on some of our samples which don't have a sign-out button, that's the way I do
- In you app, provide a sign-out button to remove the user. [PublicClientApplication](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Client-Applications#publicclientapplication) has a `Users` collection, and a `Remove(IUser)` method.

If you are happy with the answer, please close the issue

**Comment by Dark-Bond** [comment]
OK, uninstalling just to force a new set of credentials is a terrible idea.
But, running through the list of stored users in the PublicClientApplication is elegant, it does work for me, and is the proper solution to this problem.
**In plain English, the PublicClientApplication takes care of all the caching implicitly, so there's no need to create your own implementation of TokenCache.  If you want to force a login from a "Change User" button or want to force a login each time the application is started, then just clear out the users in the PublicClientApplication instance.**
Please feel free to close this issue.
